src/mono/wasm/runtime/debug.ts

```diff
@@ -46,6 +47,19 @@ export function mono_wasm_add_dbg_command_received(res_ok: boolean, id: number,
     commands_received = buffer_obj;
 }
 
+export function mono_wasm_add_dbg_command_result(res_ok: boolean, id: number, buffer: number, buffer_len: number): void {
+    const assembly_data = new Uint8Array(Module.HEAPU8.buffer, buffer, buffer_len);
+    const base64String = toBase64StringImpl(assembly_data);
+    const buffer_obj = {
+        res_ok,
+        res: {
+            id,
+            value: base64String
+        }
+    };
+    commands_result = buffer_obj;
```

Comment:
did you consider doing something like:

```suggestion
    commands_result[id] = buffer_obj;
```
Then read clear the result after for that id in the caller?